core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -33,23 +38,50 @@ public TransactionSerializer(final ArtifactSigner signer, final long chainId) {
   }
 
   public String serialize(final Transaction transaction) {
-    final byte[] bytesToSign = transaction.rlpEncode(chainId);
+    if (transaction.isEip1559()) {
+      return toHexString(serializeEip1559(transaction));
+    } else {
+      return toHexString(serializeFrontier(transaction));
+    }
+  }
+
+  private byte[] serializeFrontier(final Transaction transaction) {
+    SignatureData preSigningSignatureData =
+        new SignatureData(longToBytes(chainId), new byte[] {}, new byte[] {});
+    byte[] bytesToSign = transaction.rlpEncode(preSigningSignatureData);
+
+    SignatureData signatureData = sign(bytesToSign);
+
+    signatureData = TransactionEncoder.createEip155SignatureData(signatureData, chainId);
+    return transaction.rlpEncode(signatureData);
+  }
+
+  private byte[] serializeEip1559(final Transaction transaction) {
+    byte[] bytesToSign = transaction.rlpEncode(null);
```

Comment:
Why we  need to get the rlpEncode of null ?

Reply:
The method signature is a little confusing. We're actually encoding the transaction. 
The parameter is the initial signatureData.
For some reason, even though we haven't signed it yet, frontier txs require this to be `new SignatureData(longToBytes(chainId), new byte[] {}, new byte[] {});` but it's null for eip1559.